### PR TITLE
Detect cleanups/v11

### PR DESCRIPTION
**Pull request by victorjulien**
Rebase and reformat #10077, add a few more minor cleanups.

Also includes #10066

### CODE REVIEW

File: src/suricata.c [hunk 1]
```diff
@@ -1396,6 +1398,9 @@ static TmEcode ParseCommandLine(int argc, char** argv, SCInstance *suri)
         {"simulate-packet-tcp-ssn-memcap", required_argument, 0, 0},
         {"simulate-packet-defrag-memcap", required_argument, 0, 0},
         {"simulate-alert-queue-realloc-failure", 0, 0, 0},
+
+        {"simulate-skip-prefilter", 0, &g_skip_prefilter, 1 },
```
Comment (jasonish) [inline src/suricata.c:1402]: Is this a simulation, or actually skipping the pre-filter?

Comment (victorjulien) [reply src/suricata.c:1402]: not a simulation. Perhaps it should use a `qa` prefix or something. It needs to be clear it's not intended to be used by actual users.

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 17160

**Comment by victorjulien** [comment]
replaced by #10094

**Pull request closed without merging**